### DISCUSSION TITLE
The newest simulation results cannot be seen

### DISCUSSION THREAD

**Discussion by RowanPeng** [Spine-project/Spine-Toolbox#2012] [Support]
Hi,

I don't know why I cannot see the newest simulation results. The simulation is completed, but in the output, I can only see my previous simulation output. Could you help me with this?

**Comment by DillonJ** [comment]
Hi @RowanPeng have you selected the alternative that corresponds to your latest run?

![image](https://user-images.githubusercontent.com/7080191/221836139-ad1bbea2-eec8-483e-a0da-7ca929477bd4.png)

**Comment by RowanPeng** [comment]
Hi @DillonJ My newest simulation output doesn't exist in this list. As you can see in the picture, I just run the simulation, but I cant access to my output for today's.
<img width="449" alt="image" src="https://user-images.githubusercontent.com/123334582/221836603-e306308a-ca1a-4614-b38a-13b0774b3321.png">

**Comment by RowanPeng** [comment]
> Hi @DillonJ My newest simulation output doesn't exist in this list. As you can see in the picture, I just run the simulation, but I cant access to my output for today's. <img alt="image" width="449" src="https://user-images.githubusercontent.com/123334582/221836603-e306308a-ca1a-4614-b38a-13b0774b3321.png">

In my time zone, it is 2023-02-28. Couldn't get access to the output for today's simulation.

**Comment by nnhjy** [comment]
Hi @RowanPeng could you share a sample DB/project which can replicate the issue? That would be very helpful.

**Comment by RowanPeng** [comment]
[Exercise_4.zip](https://github.com/spine-tools/Spine-Toolbox/files/10860024/Exercise_4.zip)

**Comment by DillonJ** [comment]
Hi @RowanPeng,

I ran your model and it looks like it's infeasible. If you look at the log output you will see this:

```
...
Window 1: 2015-01-01T00:00~>2016-01-01T00:00
Optimizing model example...  3.525239 seconds (7.39 M allocations: 408.314 MiB, 3.00% gc time, 99.05% compilation time)
**Unable to find solution (reason: INFEASIBLE_OR_UNBOUNDED)**
...
```

I had a quick look at your model and here are some thoughts:

 - You could define `node_slack_penalty` for your demand node. This will mean that the model can violate the node balance at a cost. It may allow the model to solve and if you see a non-zero value for `node_slack_neg` and/or `node_slack_pos`, it may help you see where the problem is
 - We recently added `initial_[variable_name]` to allow specification of intial values for variables that are relative to the `model_start` so you don't have to use `fix_[variable_name]`. You will need to pull the latest SpineOpt to have access to this. In your case, you can replace `fix_node_state` and it's associated timeseries parameter with `initial_node_state` and a scalar parameter representing the initial state of the storage.

I hope some of that helps

**Comment by RowanPeng** [comment]
oh I see. Thank you so much.

Btw could you tell me where I can find the log output, please? Sometimes the simulation fails but I don't know what's going on.

**Reply by DillonJ** [reply]
If you look at your run_spineopt_call in your run_spineopt tool you will see:

`m = run_spineopt(ARGS...;)`

You just need to add the "log_file_path" argument as follows:

`m = run_spineopt(ARGS...; log_file_path="c:\\path\\to\\your\\file")`

Note:
 - replace c:\\path\\to\\your\\file with your actual path
 - in Julia we need to escape the `\` character so `\` becomes `\\`

There's more info here:

https://spine-tools.github.io/SpineOpt.jl/latest/library/#SpineOpt.run_spineopt

**Reply by DillonJ** [reply]
By the way - this is more of SpineOpt discussion at this point

**Reply by RowanPeng** [reply]
OK, I see. Thank you so much!
